types/react-native-text-input-mask/index.d.ts

```diff
@@ -1,16 +1,21 @@
 // Type definitions for react-native-text-input-mask 0.7
 // Project: https://github.com/react-native-community/react-native-text-input-mask
 // Definitions by: Rodrigo Weber <https://github.com/RodrigoAWeber>
+//                 Tom Xu <https://github.com/hengkx>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-import * as React from "react";
-import * as ReactNative from "react-native";
+import * as React from 'react';
+import * as ReactNative from 'react-native';
 
 export type onChangeTextCallback = (formatted: string, extracted?: string) => void;
 
-export interface TextInputMaskProps extends ReactNative.ViewProps, ReactNative.TextInputIOSProps, ReactNative.TextInputAndroidProps, ReactNative.AccessibilityProps {
-    refInput?: React.RefObject<ReactNative.TextInput>;
+export interface TextInputMaskProps
+    extends ReactNative.ViewProps,
+        ReactNative.TextInputIOSProps,
+        ReactNative.TextInputAndroidProps,
+        ReactNative.AccessibilityProps {
+    refInput?: (ref: any) => void;
```

Comment:
I'm not familiar with this library, but maybe it's supposed to be `(ref: React.RefObject<ReactNative.TextInput>) => void)`?

https://github.com/react-native-community/react-native-text-input-mask/blob/faf9a4ae6fd4e788396070542f64ffed17db79f1/index.js#L55

Reply:
ref is instance.  not React.RefObject.
https://github.com/react-native-community/react-native-text-input-mask/blob/faf9a4ae6fd4e788396070542f64ffed17db79f1/index.js#L53
```
React.ClassAttributes<TextInputMask>.ref?: string | ((instance: TextInputMask | null) => void) | React.RefObject<TextInputMask> | null | undefined
```

Reply:
And it can be of any type?  It's not somehow related to TextInput?

Reply:
Actually TextInput. I tried to write but the error did not know why

Reply:
`refInput?: (ref: ReactNative.TextInput) => void;` compiles for me.  When running the tests locally, I did have to `npm install` in `types/react`, which was somewhat non-obvious.  Maybe give that a shot?

Reply:
I've already used [react-native-masked-text](https://github.com/benhurott/react-native-masked-text) replacement.
It's not convenient to provide error screenshots.
https://github.com/benhurott/react-native-masked-text/blob/master/index.d.ts#L59